engines/wintermute/base/scriptables/script_ext_steam_api.cpp

```diff
@@ -0,0 +1,257 @@
+ *
+ */
+
+/*
+ * This file is based on WME Steam Plugin.
+ * https://bitbucket.org/MnemonicWME/wme_steam_plugin
+ * Copyright (c) 2013 Jan Nedoma
+ */
+
+#include "engines/metaengine.h"
+#include "engines/wintermute/wintermute.h"
+#include "engines/wintermute/base/base_game.h"
+#include "engines/wintermute/base/base_engine.h"
+#include "engines/wintermute/base/scriptables/script_stack.h"
+#include "engines/wintermute/base/scriptables/script_value.h"
+#include "engines/wintermute/base/scriptables/script_ext_steam_api.h"
+
+namespace Wintermute {
+
+IMPLEMENT_PERSISTENT(SXSteamAPI, false)
+
+BaseScriptable *makeSXSteamAPI(BaseGame *inGame, ScStack *stack) {
+	return new SXSteamAPI(inGame,  stack);
+}
+
+//////////////////////////////////////////////////////////////////////////
+SXSteamAPI::SXSteamAPI(BaseGame *inGame, ScStack *stack) : BaseScriptable(inGame) {
+	stack->correctParams(0);
+	init();
+}
+
+//////////////////////////////////////////////////////////////////////////
+void SXSteamAPI::init() {
+	MetaEngine &meta = ((WintermuteEngine *)g_engine)->getMetaEngine();
+	const Common::String target = BaseEngine::instance().getGameTargetName();
+	_achievementsInfo = meta.getAchievementsInfo(target);
+
+	if (_achievementsInfo.appId != "") {
+		AchMan.setActiveDomain(Common::STEAM_ACHIEVEMENTS, _achievementsInfo.appId);
+	} else {
+		warning("Unknown game accessing SteamAPI. All achievements will be ignored.");
+		AchMan.unsetActiveDomain();
+	}
+}
+
+
+//////////////////////////////////////////////////////////////////////////
+SXSteamAPI::~SXSteamAPI() {
+}
+
+
+//////////////////////////////////////////////////////////////////////////
+const char *SXSteamAPI::scToString() {
+	return "[steamapi object]";
+}
+
+
+//////////////////////////////////////////////////////////////////////////
+bool SXSteamAPI::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisStack, const char *name) {
+	//////////////////////////////////////////////////////////////////////////
+	// RequestStats()
+	// There are currently no known games that are using this
+	// So, all the initialization should be done at the constructor instead
+	//////////////////////////////////////////////////////////////////////////
+	if (strcmp(name, "RequestStats") == 0) {
+		stack->correctParams(0);
+		stack->pushBool(AchMan.isReady());
+		return STATUS_OK;
+	}
+	//////////////////////////////////////////////////////////////////////////
+	// SetAchievement(string id)
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "SetAchievement") == 0) {
+		stack->correctParams(1);
+		const char *id = stack->pop()->getString();
+		stack->pushBool(AchMan.setAchievement(id));
+		return STATUS_OK;
+	}
+	//////////////////////////////////////////////////////////////////////////
+	// IsAchieved(string id)
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "IsAchieved") == 0)
+	{
```

Comment:
Please move the curly bracket to the previous line. Same for the cases below.

Reply:
Fixed at 6da6a3d3e9fe317a12d90b26f478b67060cb41dc